gixy/parser/nginx_parser.py

```diff
@@ -50,45 +50,51 @@ def parse(self, content, root=None, path_info=None):
             #  Were parse nginx dump
             LOG.info("Switched to parse nginx configuration dump.")
             root_filename = self._prepare_dump(parsed)
+            path_info = root_filename
             self.is_dump = True
             self.cwd = os.path.dirname(root_filename)
             parsed = self.configs[root_filename]
 
-        self.parse_block(parsed, root)
+        self.parse_block(parsed, root, path_info)
         return root
 
-    def parse_block(self, parsed_block, parent):
+    def parse_block(self, parsed_block, parent, path_info):
```

Comment:
Every method is now exploded with path_info. Instead, can we push the filename into a stack, e.g. add `self._path_stack = []` in init, add it in `parse_file` with `self._path_stack.append(path)` and use it everywhere needed with:

```
    @property
    def path_info(self):
        """Current file being parsed, or None."""
        return self._path_stack[-1] if self._path_stack else None
  ```

Reply:
Not sure if that'll work considering the `include` handling

Reply:
But something similar works, yep

Reply:
https://github.com/dvershinin/gixy/pull/36/commits/bbc5072333b8389d80dc0b2e437d0081ac7e55a1

Reply:
no, that doesn't work because when an include is used, the original filename is not set from the includer. fixed in https://github.com/dvershinin/gixy/pull/36/commits/0feefb3a06dadeb3f7ade4ee0a5f6ec1ce287dc3